src/mpid/ch4/shm/src/shm_p2p.h

```diff
@@ -159,12 +179,12 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_SHM_mpi_isend(const void *buf, MPI_Aint count
     size_t data_sz;
 
     MPIDI_Datatype_check_contig_size(datatype, count, dt_contig, data_sz);
-    if (MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE > -1 &&
-        dt_contig && data_sz > MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE) {
```

Comment:
You should not delete the `MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE > -1 ` condition check.

Reply:
I think we should move `MPIR_Datatype_add_ref_if_not_builtin(datatype);` into `MPIDI_XPMEM_lmt_rts_isend`.

Reply:
I may delete `MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE > -1 ` by accident, will add it back.